Rename poorly named tests in `Zend/tests`

alexandre-daubois opened the pull request.

Girgias reviewed (approved):
Wonder if some should be moved into a folder (e.g. the closure ones into the `tests/closures/` folder

alexandre-daubois commented on Zend/tests/func_get_arg_basic.phpt:
```diff
@@ -104,7 +104,7 @@ int(10)
 int(1)
 func_get_arg(): Argument #1 ($position) must be less than the number of the arguments passed to the currently executed function
-Exception: Too few arguments to function test2(), 0 passed in %s002.php on line %d and exactly 1 expected
+Exception: Too few arguments to function test2(), 0 passed in %sfunc_get_arg_basic.php on line %d and exactly 1 expected
```
Right!

alexandre-daubois commented:
> Wonder if some should be moved into a folder

I agree. I moved some of the tests I renamed to directories when relevant. While at it, I definitely saw more files to move. Ultimately, everything should be in a directory I guess?

DanielEScherzer reviewed (approved):
Its probably going to bug me that the numbered tests (Zend/tests/%d.phpt) are now no longer consecutive due to gaps, but thats fine for now.
